client/core/core.go

```diff
@@ -3395,33 +3410,40 @@ func (c *Core) initialize() {
 	}
 }
 
-// verifyAccount verifies AccountInfo by making a connection to the DEX.
-func (c *Core) verifyAccount(acct *db.AccountInfo) bool {
+// connectAccount makes a connection to the DEX for the given account. If a
+// non-nil dexConnection is returned, it was inserted into the conns map even if
+// it is the initial connection attempt failed (connected == false); the connect
+// retry / keepalive loop is active.
+func (c *Core) connectAccount(acct *db.AccountInfo) (dc *dexConnection, connected bool) {
+	if !acct.Paid && len(acct.FeeCoin) == 0 {
+		// Register should have set this when creating the account that was
+		// obtained via db.Accounts.
+		c.log.Warnf("Incomplete registration without fee payment detected for DEX %s. "+
+			"Discarding account.", acct.Host)
+		return
+	}
+
 	host, err := addrHost(acct.Host)
 	if err != nil {
-		c.log.Errorf("skipping loading of %s due to address parse error: %v", acct.Host, err)
-		return false
+		c.log.Errorf("skipping loading of %s due to address parse error: %v", host, err)
+		return
 	}
-	dc, err := c.connectDEX(acct)
-	if err != nil {
-		c.log.Errorf("error connecting to DEX %s: %v", acct.Host, err)
-		return false
+	dc, err = c.connectDEX(acct)
+	if dc == nil {
```

Comment:
@buck54321 had a good point in another thread about using dc.acct.host instead of doing the superfluous `addrHost` call a few lines up since `connectDEX` does it too.

Reply:
aparently the rebase changed it